### PR TITLE
BG-12345 extend migrated bch tool

### PR DESCRIPTION
**Pull request by dannybitgo**
https://bitgoinc.atlassian.net/browse/BG-12345

Extend the Migrated-BCH tool to include BTG and BSV

### CODE REVIEW

File: src/components/migrated-legacy.js [hunk 1]
```diff
@@ -8,11 +8,13 @@ import * as _ from 'lodash';
...
 import tooltips from '../constants/tooltips';
+import coinConfig from 'constants/coin-config';
 
-const formTooltips = tooltips.migratedBch;
+const formTooltips = tooltips.migratedLegacy;
 
-class MigratedBchRecoveryForm extends Component {
+class MigratedRecoveryForm extends Component {
   state = {
+    coin: 'bch',
```
Comment (cooncesean) [inline src/components/migrated-legacy.js]: So what's up... by default, this class is configured to migrate BCH, but can be overridden if specified by the user?

File: src/components/migrated-legacy.js [hunk 2]
```diff
@@ -155,9 +161,9 @@ class MigratedBchRecoveryForm extends Component {
     const { bitgo } = this.props;
     this.setState({ error: '', recovering: true });
 
-    const bch = bitgo.coin('bch');
-    const bchWallets = await bch.wallets().list();
-    const migratedWallet = _.find(bchWallets.wallets, w => w._wallet.migratedFrom === this.state.walletId);
+    const coin = bitgo.coin(this.props.bitgo.env === 'prod' ? this.state.coin : `t${this.state.coin}`);
+    const wallets = await coin.wallets().list();
+    const migratedWallet = _.find(wallets.wallets, w => w._wallet.migratedFrom === this.state.walletId);
```
Comment (cooncesean) [inline src/components/migrated-legacy.js]: is there a cleaner way to do this?  ie: why fetch the whole list when all we're looking for is a single `migratedWallet`? 

could this be condensed into something like:

```javascript
migratedWallet = await coin.wallets(migratedFrom=this.state.WalletId);
```
(i'm not familiar w/ the orm to know if the above will actually work - but you get the gist)

File: src/constants/coin-config.js [hunk 3]
```diff
@@ -113,6 +113,10 @@ export default {
   supportedRecoveries: {
     crossChain: ['btc', 'bch', 'ltc', 'bsv'],
     nonBitGo: ['btc', 'eth', 'xrp', 'bch', 'xlm', 'ltc', 'dash', 'zec', 'btg', 'token'],
-    unsignedSweep: ['btc', 'bch', 'ltc', 'xrp', 'xlm', 'dash', 'zec', 'btg', 'eth', 'token']
+    unsignedSweep: ['btc', 'bch', 'ltc', 'xrp', 'xlm', 'dash', 'zec', 'btg', 'eth', 'token'],
+    migrated: {
+      test: ['bch', 'bsv'],
```
Comment (cooncesean) [inline src/constants/coin-config.js]: should these be `tbch`, `tbsv`?

File: src/components/migrated-legacy.js (hunk 2 above)
Comment (dannybitgo) [reply src/components/migrated-legacy.js]: so, ```state.walletId``` is actually the v1 BTC wallet. We could do something like what you mentioned, but that would require us to add an additional field in the UI (or change the current field in the UI). It actually might make more sense to change the current field in the UI. I'll take a look tomorrow

File: src/constants/coin-config.js (hunk 3 above)
Comment (dannybitgo) [reply src/constants/coin-config.js]: The way the code is written it works as is. You can see none one of the other ones use the 't' prefix as all the other tools append that on in front for testnet

File: src/components/migrated-legacy.js (hunk 1 above)
Comment (dannybitgo) [reply src/components/migrated-legacy.js]: yes exactly

Comment (dannybitgo) [reply src/components/migrated-legacy.js]: so BCH is the default for the drop-down menu

**Comment by dannybitgo** [comment]
Update, this exact version has successfully recovered BSV and BTG

**Comment by CapnMigraine** [comment]
I tested this PR with Danny, we were able to recover my 3 migrated wallets (1 BCH, 1 BTG and 1 BSV) running the wrw with these changes.

**Pull request merged**